src/main/java/bio/terra/cli/context/utils/Logger.java

```diff
@@ -0,0 +1,184 @@
+            .resolve("%i." + GlobalContext.getLogFile().getFileName())
+            .toString());
+    rollingPolicy.setParent(rollingFileAppender);
+
+    // the base file name (e.g. terra.log) is not included in rollingPolicy's count, so subtract one
+    // here to account for that
+    // e.g. MAX_NUM_FILES=3: terra.log, 1.terra.log, 2.terra.log
+    if (MAX_NUM_FILES < 1) {
+      throw new IllegalArgumentException("Maximum number of log files must be >= 1");
+    }
+    rollingPolicy.setMinIndex(1);
+    rollingPolicy.setMaxIndex(MAX_NUM_FILES - 1);
+    rollingFileAppender.setRollingPolicy(rollingPolicy);
+
+    // make sure to start the policies after the cross-references between the policy and appender
+    // have been set
+    triggeringPolicy.start();
+    rollingPolicy.start();
+
+    setupEncoderAndFilter(
+        rollingFileAppender, loggerContext, globalContext.fileLoggingLevel.getLogLevelImpl());
+    rollingFileAppender.start();
+
+    // build the console appender
+    ConsoleAppender consoleAppender = new ConsoleAppender();
+    consoleAppender.setName("ConsoleAppender");
+    consoleAppender.setContext(loggerContext);
+    setupEncoderAndFilter(
+        consoleAppender, loggerContext, globalContext.consoleLoggingLevel.getLogLevelImpl());
+    consoleAppender.start();
+
+    // on the root logger, clear any existing appenders and attach the two created above
+    ch.qos.logback.classic.Logger rootLogger = loggerContext.getLogger(ROOT_LOGGER_NAME);
+    rootLogger.detachAndStopAllAppenders();
+    rootLogger.addAppender(rollingFileAppender);
+    rootLogger.addAppender(consoleAppender);
+
+    // if a process is too short-lived, then the policy may not check if it should rollover.
+    // each CLI command invocation is a new process (i.e. new JVM) and may be very short-lived.
+    // so, include an additional manual check here on startup, to see if we should roll over the
+    // file based on its size.
+    File activeFile = new File(rollingPolicy.getActiveFileName());
+    if (activeFile.length() > MAX_FILE_SIZE) {
+      rollingFileAppender.rollover();
+    }
+
+    // StatusPrinter.print(loggerContext); // helpful for debugging
+  }
+
+  /**
+   * Helper method to setup the encoder and filter for the given appender. This setup is common to
+   * all appenders.
+   *
+   * @param appender appender object to setup
+   * @param loggerContext global logger context object
+   * @param loggingLevel logging level particular to this appender
+   */
+  private static void setupEncoderAndFilter(
+      OutputStreamAppender appender, LoggerContext loggerContext, Level loggingLevel) {
+    PatternLayoutEncoder encoder = new PatternLayoutEncoder();
+    encoder.setContext(loggerContext);
+    encoder.setPattern(LOG_FORMAT);
+    encoder.start();
+    appender.setEncoder(encoder);
+
+    // filter out any logs that are below the fileLoggingLevel specified in the global context
+    appender.clearAllFilters();
+    appender.addFilter(
```

Comment:
nit / side note: I'm a little surprised there isn't some built-in class that does this already.

Reply:
You're right. Found `ThresholdFilter` and used that instead.